### ISSUE TITLE
Time line as a tool for working with outdated point on the map

### ISSUE THREAD

**Issue by rizomaa** [diglabby/mapa#65]
The more rare initiative updates the more it points become transparent.

**Comment by wellemut** [comment]
That is a good idea.
I would suggest, that an entry is only half visible after 1,5 years without
editing and will be no more visible after 2 years.
Except is someone searches for this entry or should be shown so the user
can update it.

But there are a few requirements:

- Every year, the entry owner will get a notification asking if he is still
alive.
- If owner does not respond for half a year, the creator or regional pilot
will be asked by mail.
- if he does not answer too, the entry will be hided after two years.

Helmut Wolman

(mailed on the move)
Mobil: 0049-15734448245
Mannheim

Mister Cable <notifications@github.com> schrieb am Mo., 12. Aug. 2019,
15:04:

**Comment by rizomaa** [comment]
@wellemut sounds as very long procedure. 

"- Every year, the entry owner will get a notification asking if he is still
alive."
what about point without direct contact like email?

**Comment by k-gorod** [comment]
Or registration is required, or points without feedback(log in) can be only like one-time event. And after finishing must be deleted. If someone want to create something more, he must to be logged. Without that we cant control map and clear trash.

**Comment by rizomaa** [comment]
@k-gorod we will follow concept as little registration is possible.

Now I have no idea about appearance of the feature, but functionally it should include
1. Possibility to observe all commits (data updates) with dates.
2. Possibility click on each commit ad observe data on the commit.

As a User story the issue can sound like:
I as a visitor of the Mapa wants to understan all key points where data af particular pin on the map was changed and what was the information there.

Does it make sense for you?

**Comment by k-gorod** [comment]
1)Calendar?  It is big work with another modal window, or full page i think. It is good idea, and big work with new design, discusses and including ideas. 
2)@rizomaa
>we will follow concept as little registration is possible.

But what about spam? What about inadequate people, who creates unacceptable content? How u ban them? Fast registation is: enter email or number, name(if user want to), confirm email or number.

**Comment by k-gorod** [comment]
I think, if i would like to find as fast as possible JS meet up, or business conference... I would not happy to see 40 points of meeting of pokemon catchers.

**Comment by rizomaa** [comment]
@k-gorod It is a task about tracking changes. It minimizes vandalism and gives a tool to observe an history of each record. As first step, the task should be clarified and may be visualized. Do you understand a common idea of the task?

About vandals. We rely on the trust and force inside the community and openness which we add into the app architecture. 

Also, there is a related issue linked to the minimization of vandalism  https://github.com/diglabby/mapa/issues/11 When you complete this one, after you can help to solve that about Captcha.

Does it make sense?

**Comment by k-gorod** [comment]
Yes, and thats why i wrote answer on first point of my message. 
if I understand correctly, you mean modal window with something like calendar(or line with dates), which would show events, changings and something like that.

**Comment by k-gorod** [comment]
Or it is like modal with news notifications, like in socials?!

**Comment by rizomaa** [comment]
![Selection_053](https://user-images.githubusercontent.com/9870374/83291118-7e375580-a1f0-11ea-813c-b0d54331628a.png)

Check list:
1. It should be the list of links (one link for each version) at the bottom part of a card.
2. When you click the link on particular version there appears on popup with the version (I guess here you can use API to extract)
3. Window can be closed by clicking on a cross sign on this pop-up or any place out side.

Any comments?

**Comment by k-gorod** [comment]
About this->

>Possibility to observe all commits (data updates) with dates.
>It should be the list of links (one link for each version) at the bottom part of a card.
>When you click the link on particular version there appears on popup with the version (I guess here >you can use API to extract)
>Window can be closed by clicking on a cross sign on this pop-up or any place out side.

I think this information must be available only to creator, and developer. Because i cant imagine average user, who looking for not actual information. And maybe some of creators don't want to someone see previous versions of cards, and it is important like privacy
I think we must create smth like button "My cards". In there create list with card of this user, like our usual list but every "precard" with button "settings"

![image](https://user-images.githubusercontent.com/58625453/83303314-8057df00-a205-11ea-9cd8-3a6aaab6b238.png)
or list of points what looking like that 
![image](https://user-images.githubusercontent.com/58625453/83303697-31f71000-a206-11ea-8662-cd70aec36ef9.png)
And in that section(settings of card) add things like u talking about(commented above).
My thinks: 
This feature is completely irrelevant now.
First of all, for correct realize of this, like i say before, we need to create session with login. We need to bind events to creators. So that I can’t, for example, write a letter "Ы" in card of another user.
![image](https://user-images.githubusercontent.com/58625453/83305165-e72ac780-a208-11ea-87d1-b744189df03b.png)
I can do it now, without any registration.

Secondly, this feature is very unimportant. imho. Or i didnt understand you again.

About issue:
I think it is important to let user know about "Relevance of event".
I suggest this algorithm:
1) @wellemut
>Every year, the entry owner will get a notification asking if he is still alive.

But every one or three month, or maybe week.

2) Create 2 states: active(actual) - not active. 

3) @rizomaa 
>The more rare initiative updates the more it points become transparent. 

After some time without answering give to card status smth like "Not active" or "Relevance not confirmed" on card. And make point on map more transparent, for example.

If user creates event without log in(i think we must disable this possibility) - it is will be deleted after month or three anyways(notify about it on creation form).

This is it. What you think?

**Comment by rizomaa** [comment]
@k-gorod thanks for the comprehensive analysis.

Point by point:
1. It is needs to be done without any registration. Here is not the "privacy" case, because all information when it is put in the Mapa is under CC0 (Public domain). Verification by version gives a strong tool to keep data integrity through the whole life of the entry.
2. Here is linked regardless type of an entry both either it is event or initiative.
3. Notification sounds reasonable, but do not touch any notification system in the issue. We will think about it a bit later.
4. Changing status (to not active etc) sounds good also, please create new card for it.
5. Registration it is also out of the task scope. 

Are the answers understandable?

PS
I also changed  the title of the issue to make it clear for understanding,

**Comment by wellemut** [comment]
Hi @rizomaa
This issue gets a bit diffuse, I hope I understand the context. 

>     1. It is needs to be done without any registration. Here is not the "privacy" case, because all information when it is put in the Mapa is under CC0 (Public domain). Verification by version gives a strong tool to keep data integrity through the whole life of the entry.
-> I think we can not make all previous versions public, because some change are made, because someone falsy used a private mailadress or mentioned a private person. Therefore only Moderators have access to the history so far. If you need access to history, please register on openfairdb.org/ and send me your mailadress. Helmut@kartevonmorgen.org

**Comment by rizomaa** [comment]
@wellemut what about licence CC0 on every added entry?

**Comment by wellemut** [comment]
I think is is a crucial point. Yes it is open source, but what about mistakes people do?
What happens if you would publish private Fotos or Contact infos on Wikipedia? Is it accessible there for ever?
But We can think about making Title, Location (at least the citiy, not the street) and hashtags visible for everyone. But even description could contain things, that you just are not allowed to share publicly.

**Comment by rizomaa** [comment]
@wellemut 

CC0 means public domain = any person can use the data for any purpose with keeping moral author rights. And users who adds the data accept the rules. Or you use other meaning for CC0 here. Please provide. 

![Selection_056](https://user-images.githubusercontent.com/9870374/84385113-b169e380-abf7-11ea-9e3e-48efd5309f3f.png)

Sure, there are bad guys but the system (I mean the map backend) has no technical restriction to do it in a perfect way now. For example, for spam... etc.

I'm convinced if there are some extra restriction in the rules but not written, they should be landed on the checkbox area. But before discussed in other issue because changing the policies goes beyond of this issue scope.

**Comment by k-gorod** [comment]
Then i dont understand u more then ever. Usual user comes on site, and looking for not actual card of events? For what?

**Comment by k-gorod** [comment]
My opinion - without binding event to creator, without cards security form another users, this site cant perform its main idea. And if we confront with that problem in that level, it will create more and more problems in future.

**Comment by rizomaa** [comment]
@k-gorod 

1. У вікіпедыі ёсць сістэма кантроля версій
https://be.wikipedia.org/w/index.php?title=%D0%9F%D1%8F%D1%81%D1%87%D0%B0%D0%BD%D0%B0%D1%8F_%D0%B1%D1%83%D1%80%D0%B0&action=history

2. Тут на старонках таксама ёсць таки функцыянал:
https://github.com/diglabby/mapa/wiki/How-to-contribute/_history

Гэты функцыянал дае магчымасць праглядзець папярэднюю версію.

Так відавочней?

**Comment by k-gorod** [comment]
Да, но дело в том, что Википедия-это энциклопедия, написанная самими людьми. И когда создали новое "описание предмета", оно может быть не настолько для всех приемлимо, и тогда это "все" будут копать глубже в поисках всей информации что есть. И возможно пожалуются и проголосуют за предедущую версию.
И тут важно понимать, что событие - это намного более узконаправленная информация. В какой ситуации может понадобиться создателю изменить свою карточку о событии? Поменялся телефон? Поменялось место. Решил изменить описание на более лучшее.

Кому нужны предыдущие версии?
-Это раз.
Во-вторых даже для Википедии - это настолько анюзабельныая фича. Возможно в один из несколько сотен миллионов запросов ей воспользуется. 
Для этого:1)человек должен придти, начать читать всю информацию
2)прочитав, ему она должна не понравиться.
3)Он должен знать, что такая фича есть(я вот например не знал, хотя очень много "википедил")
4)и лишь тогда, если прям важно что-то найти, он полезет этим пользоваться.

Мы говорим уже о научном подходе, дотошной точности в получаемой информации, когда говорим о этой функции в Википедии, и я уверен, что она была реализована гараздо позднее того, как сайт выкатили, он стал известным и т.д.

Как я и говорил. В нашем случае такой функционал возможно потребуется только создателю, и то в редчайших случаях.

Вот например даже в Википедии это не так и нужно, как например здесь-в ГитХабе. Вы можете сейчас нажать, и посмотреть предыдущие версии того что я сказал. Таким образом, изменив свое сообщение, я уже не могу не отвечать за то, что сказал до изменения, и так далее...

**Comment by rizomaa** [comment]
@k-gorod 
дзякуй за тваё меркаванне. 

На сёння:
1. нам важна рабіць як мага больш фіч, якія рэалізуюць празрыстасць і ўстойлівасць дадзеных грамадскіх ініцыятыў пабудаваных на даверы супольнасці.
2. у той жа час важна стварыць механізм верыфікуемасці і зменшыць уплыў негатыўных фактараў (спам, падмена дадзеных) без стварэння залішняй інфраструктуры рэгістрацыі, або мадэратараў.

Калі лічыш, што задачу тэхнічна не пад сілах вырашыць, або калі на твой погляд яна не мэтазгодна, напішы. Мы яе перавызначым на іншых распрацоўшчыкаў.

У любым разе, ты дапамог палепшыць яе апісанне.

**Comment by k-gorod** [comment]
Да не вопрос. Дайте мне API запросы на гет и пост в базу данных json объектов хранящих все это. И я это сделаю.

**Comment by rizomaa** [comment]
@k-gorod тут можаш прачытаць інструкцыю як адкрымаць юсу свежую дакументацыю па АПІ https://github.com/diglabby/mapa/issues/249#issue-535775346
